src/tests/test_sexual_behaviour.py

```diff
@@ -23,10 +24,12 @@ def check_prob_sums(sex, trans_matrix):
 
 
 def test_transition_probabilities():
-    for trans_matrix in sbd.sex_behaviour_trans_male_options:
+    with open("data/sex_behaviour.yaml",'r') as file:
+        yaml_data = yaml.safe_load(file)
```

Comment:
That's true, I think the reason I tested the file instead is because this problem is arguably independent of the object - it's checking for invalid data in the input file, and so if it failed because there was a problem with, for example, the object failing to initialise properly, then that's kind of a separate issue. Might be worth looking at again in the future, although I think this test will become redundant as the probabilities will get internally renormalised.